x-pack/plugin/security/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -59,3 +59,9 @@ grant codeBase "${codebase.httpasyncclient}" {
   // rest client uses system properties which gets the default proxy
   permission java.net.NetPermission "getProxySelector";
 };
+
+grant codeBase "${codebase.mina-core}" {
+   // kerberos test simple kdc server component depends on apache mina,
+   // which internally requires this permission
+   permission java.lang.RuntimePermission "accessClassInPackage.sun.reflect";
```

Comment:
This adds the permission to production too, yet this is only for a test. I understand that mina is a test-only dependency. Is there anything that we can do to avoid this?

Reply:
+1 to not modifying production permissions. 

The failure is in a test for a test class. What about moving these classes to their own project where the permission can be added and then security has a test dependency on this project? This limits the scope of the permission and security is tested without additional permissions.

Reply:
@bizybot Can you link to the offending Apache Mina code?

Reply:
Note that we have evil-tests that disable the security manager in tests, that could be an option here too. I would like to see the Apache Mina code first, to see if we can fix this upstream or through an outrageous hack.

Reply:
Thanks so much for that, this helps a lot. The critical loop is this:

```
        for (StackTraceElement s : stackTrace) {
            try {
                Class<?> cls = DefaultIoFuture.class.getClassLoader().loadClass(s.getClassName());
                
                if (IoProcessor.class.isAssignableFrom(cls)) {
                    throw new IllegalStateException("DEAD LOCK: " + IoFuture.class.getSimpleName()
                            + ".await() was invoked from an I/O processor thread.  " + "Please use "
                            + IoFutureListener.class.getSimpleName()
                            + " or configure a proper thread model alternatively.");
                }
            } catch (ClassNotFoundException cnfe) {
                // Ignore
            }
        }
```

They already account for CNFE here. They should account for SME too, if they can not load the class they should assume that it is not an `IoProcessor` implementation and skip it like they skip CNFE. Can we contribute this upstream please? Then we need a `mina` release, then we need a `ldap-directory-api` release incorporating that. Can you own this @bizybot?

In the meantime, let us make this an evil test since it will take some time for the above to take place.